syntax = "proto3";

import "AccountService.proto";
import "GameCharacterService.proto";

message DatabaseRequestWrapper {
    uint32 session_id = 1;
    oneof service_request {
        AccountServiceRequestWrapper account_service_request = 2;
        GameCharacterServiceRequestWrapper gamecharacter_service_request = 3;
    }
}

message DatabaseResponseWrapper{
    uint32 session_id = 1;

    oneof service_response {
        AccountServiceResponseWrapper account_service_response = 2;
        GameCharacterServiceResponseWrapper gamecharacter_service_response = 3;
    }
}
